### PR TITLE
[ENH] use multiplatform compatible regular expressions

### PR DESCRIPTION
**Pull request by MaxvandenBoom**
Hi @sappelhoff ,

This PR allows the regular expressions to work with both unix and windows filepaths, see https://github.com/bids-standard/bids-validator/issues/959.

I updated all the regexes, however the top-level files fail in the validation because these (as specified in `fixed_top_level_names.json`) are evaluated as literal strings (e.g. `"/participants.tsv"`), rather than regular expressions.

There is a file `top_level_rules.json` besides `fixed_top_level_names.json`, so I could adjust the code that all top-level files are evaluated by regexes, but maybe there is a good reason why they are in a seperate `fixed_` file. What do you think?

### CODE REVIEW

**Comment by sappelhoff** [comment]
Thanks for sending the PR over!

> but maybe there is a good reason why they are in a seperate fixed_ file

maybe, I am not sure --> Can you please help us on this one as well @nellh ? or @rwblair ?

From looking at it I think that regexps would make more sense.

**Comment by MaxvandenBoom** [comment]
Little bump for attention. Just say the word and can fix it...

**Comment by rwblair** [comment]
As best I can tell its a convenience thing, switching them over to be regex shouldn't be problematic. Besides writing the regex I think the only three places that will need to be updated to look more like its regex based siblings are:

https://github.com/bids-standard/bids-validator/blob/8891c2978c41aeac28d89324baadc9b5be76365b/bids-validator/bids_validator/bids_validator.py#L77

https://github.com/bids-standard/bids-validator/blob/8891c2978c41aeac28d89324baadc9b5be76365b/bids-validator/utils/type.js#L50

https://github.com/bids-standard/bids-validator/blob/8891c2978c41aeac28d89324baadc9b5be76365b/bids-validator/utils/type.js#L95

**Comment by MaxvandenBoom** [comment]
Ok! Then I'll update this pull request and we can review

**Comment by sappelhoff** [comment]
Just a reminder that we are still very interested in this PR (whenever you have some time to spare) @MaxvandenBoom :slightly_smiling_face:

**Comment by MaxvandenBoom** [comment]
Haha, I haven't forgotten and good to hear it would help the project. Was planning to look into it this weekend :)

File: bids-validator/utils/type.js [hunk 1]
```diff
@@ -92,7 +91,7 @@ export default {
      */
     isTopLevel: function(path) {
       return (
-        fixedTopLevelNames.indexOf(path) != -1 ||
+		rootTop.test(path) ||
         funcTop.test(path) ||
```
Comment (sappelhoff) [inline bids-validator/utils/type.js]:
```suggestion
        rootTop.test(path) ||
        funcTop.test(path) ||
```

File: bids-validator/bids_validator/rules/top_level_rules.json [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+        "CHANGES",
+		"LICENSE",
+		"dataset_description.json",
+		"genetic_info.json",
+		"participants.tsv",
+		"participants.json",
+		"phasediff.json",
+		"phase1.json",
+		"phase2.json",
+        "fieldmap.json"
```
Comment (sappelhoff) [inline bids-validator/bids_validator/rules/top_level_rules.json]:
```suggestion
        "README",
        "CHANGES",
        "LICENSE",
        "dataset_description.json",
        "genetic_info.json",
        "participants.tsv",
        "participants.json",
        "phasediff.json",
        "phase1.json",
        "phase2.json",
        "fieldmap.json"
```

**Review by sappelhoff — COMMENTED** [review]
thanks Max, that looks good to me so far!

**Review by sappelhoff — APPROVED** [review]
LGTM!

**Comment by MaxvandenBoom** [comment]
That's quick :)
Thanks for the suggestions, it works for me like this

**Comment by rwblair** [comment]
@MaxvandenBoom Thank you for all the changes!

**Pull request merged**